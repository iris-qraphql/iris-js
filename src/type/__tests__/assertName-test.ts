import { assertEnumValueName, assertName } from '../assertName';

describe('assertName', () => {
  it('passthrough valid name', () => {
    expect(assertName('_ValidName123')).toEqual('_ValidName123');
  });

  it('throws for non-strings', () => {
    expect(() => assertName({} as any)).toThrowErrorMatchingSnapshot();
  });

  it('throws on empty strings', () => {
    expect(() => assertName('')).toThrowErrorMatchingSnapshot();
  });

  it('throws for names with invalid characters', () => {
    expect(() => assertName('>--()-->')).toThrowErrorMatchingSnapshot();
  });

  it('throws for names starting with invalid characters', () => {
    expect(() => assertName('42MeaningsOfLife')).toThrowErrorMatchingSnapshot();
  });
});

describe('assertEnumValueName', () => {
  it('passthrough valid name', () => {
    expect(assertEnumValueName('_ValidName123')).toThrowErrorMatchingSnapshot();
  });

  it('throws on empty strings', () => {
    expect(() => assertEnumValueName('')).toThrowErrorMatchingSnapshot();
  });

  it('throws for names with invalid characters', () => {
    expect(() =>
      assertEnumValueName('>--()-->'),
    ).toThrowErrorMatchingSnapshot();
  });

  it('throws for names starting with invalid characters', () => {
    expect(() =>
      assertEnumValueName('42MeaningsOfLife'),
    ).toThrowErrorMatchingSnapshot();
  });

  it('throws for restricted names', () => {
    expect(() => assertEnumValueName('true')).toThrowErrorMatchingSnapshot();
    expect(() => assertEnumValueName('false')).toThrowErrorMatchingSnapshot();
    expect(() => assertEnumValueName('null')).toThrowErrorMatchingSnapshot();
  });
});
