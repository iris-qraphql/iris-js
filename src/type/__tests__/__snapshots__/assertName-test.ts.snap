// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertEnumValueName passthrough valid name 1`] = `"received is not a function"`;

exports[`assertEnumValueName throws for names starting with invalid characters 1`] = `"Names must start with [_a-zA-Z] but \\"42MeaningsOfLife\\" does not."`;

exports[`assertEnumValueName throws for names with invalid characters 1`] = `"Names must only contain [_a-zA-Z0-9] but \\">--()-->\\" does not."`;

exports[`assertEnumValueName throws for restricted names 1`] = `"Enum values cannot be named: true"`;

exports[`assertEnumValueName throws for restricted names 2`] = `"Enum values cannot be named: false"`;

exports[`assertEnumValueName throws for restricted names 3`] = `"Enum values cannot be named: null"`;

exports[`assertEnumValueName throws on empty strings 1`] = `"Expected name to be a non-empty string."`;

exports[`assertName throws for names starting with invalid characters 1`] = `"Names must start with [_a-zA-Z] but \\"42MeaningsOfLife\\" does not."`;

exports[`assertName throws for names with invalid characters 1`] = `"Names must only contain [_a-zA-Z0-9] but \\">--()-->\\" does not."`;

exports[`assertName throws for non-strings 1`] = `"Expected name to be a string."`;

exports[`assertName throws on empty strings 1`] = `"Expected name to be a non-empty string."`;
